app/models/dns-record.server.ts

```diff
@@ -36,6 +37,10 @@ export function getUserDnsRecordCount(username: DnsRecord['username']) {
 export async function createDnsRecord(
   data: Required<Pick<DnsRecord, 'username' | 'type' | 'subdomain' | 'value'>> & Partial<DnsRecord>
 ) {
+  if (await isDeactivated(data.username)) {
```

Comment:
`isUserDeactivated` would be a better name I think.

Reply:
Yeah, that is better. I can update this